iosxr_iso2vbox.py

```diff
@@ -296,130 +320,140 @@ def cleanup_vms(name, box_name):
     run('VBoxManage storageattach %s --storagectl IDE_Controller --port 1 --device 0 --type dvddrive --medium %s' % (vmname, input_iso))
 
     # Change boot order to hd then dvd
-    verboseprint('Boot order disk first')
+    logger.debug('Boot order disk first')
     run('vboxmanage modifyvm %s --boot1 disk' % vmname)
 
-    verboseprint('Boot order DVD second')
+    logger.debug('Boot order DVD second')
     run('vboxmanage modifyvm %s --boot2 dvd' % vmname)
 
     # Print some information about the VM
-    verboseprint('VM Info:')
+    logger.debug('VM Info:')
     run('VBoxManage showvminfo %s' % vmname)
 
-    # Start the VM for installation of ISO - must be started in the background
-    verboseprint('Starting VM...')
-    cmdstring = ('VBoxHeadless --startvm %s' % vmname)
-    args = shlex.split(cmdstring)
-    subprocess.Popen(args)
-    time.sleep(2)
+    # Start the VM for installation of ISO - must be started as a sub process
+    logger.debug('Starting VM...')
+    cmd_string = ('VBoxHeadless --startvm %s' % vmname)
+    run_process_quietly(cmd_string)
 
     while True:
-        vms_list = str(run('VBoxManage showvminfo %s' % vmname))
+        vms_list = str(run_get_output('VBoxManage showvminfo %s' % vmname))
         if 'running (since' in vms_list:
-            verboseprint('Successfully started to boot VM disk image')
+            logger.debug('Successfully started to boot VM disk image')
             break
         else:
-            verboseprint('Failed to install VM disk image\n')
+            logger.debug('Failed to install VM disk image\n')
             continue
 
     # Use iosxr_pexpect.py to bring up XR and do some initial config.
     # Using socat to do the connection as telnet has an
     # odd double return on vbox
-    verboseprint('Bringing up with iosxr_pexpect.py to install to disk and configure')
+    logger.info('Bringing up with iosxr_pexpect.py to install to disk and configure')
     iosxr_pexpect_path = os.path.join(pathname, 'iosxr_pexpect.py')
-    cmdstring = "python %s -cmds 'socat TCP:localhost:%s -,raw,echo=0,escape=0x1d' -config iosxr_setup" % (iosxr_pexpect_path, console_port)
-    subprocess.call(cmdstring, shell=True)
-    # Todo: this isn't suppressing the output
-    # subprocess.call(cmdstring, shell=True, stdout=open(os.devnull, 'wb'))
+    if verbose is True:
+        verbose_str = '-v'
+        verbose_pipe = ''
+    else:
+        # Shhhh...
+        verbose_str = ''
+        verbose_pipe = '>/dev/null 2>&1'
+
+    cmd_string = "python %s %s -cmds 'socat TCP:localhost:%s -,raw,echo=0,escape=0x1d' -config iosxr_setup" % (verbose_str, iosxr_pexpect_path, console_port)
+    # Call socat connection from a script so it can be run in the background
+    f = open('runme.sh', 'w')
+    f.write(cmd_string)
+    f.close()
+    rc = subprocess.call("chmod 766 ./runme.sh; ./runme.sh %s" % verbose_pipe, shell=True)
+    if rc == 0:
+        sys.exit('Configuring XR failed, exiting')
```

Comment:
This is the convoluted nature of iso2vbox->pexpect->setup true. Yes by background I meant silently. Working on this.